demos/projects/ESPRESSIF/adu/README.md

```diff
@@ -1,30 +1,296 @@
-# ADU Update
+# Perform an Over the Air Update with the ESP32
 
-## ESP32 Board Directions
+This sample will allow you to update an ESP32 over the air (OTA) using Azure Device Update. The following is an outline of the steps to run this sample.
 
-- **NOTE BEFORE ADVANCING**: [The image to download is currently hardcoded here.](https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/7e419d65d0dea0ae5ea5107cc50b2a8f9d407f12/libs/azure-iot-middleware-freertos/source/azure_iot_adu_client.c#L257-L264) If you would like to use your own image, upload and provide the links there. Otherwise, the image there will print out a "Version 2" to the console. The default image will likely have different WiFi credentials so it won't connect once it reboots.
-- Build the image with the next version: [LINK HERE](https://github.com/Azure-Samples/iot-middleware-freertos-samples/blob/1f63c83e2642203fa003b74067a7421f76ad1cfc/demos/sample_azure_iot_adu/sample_azure_iot_adu.c#L160)
-- Upload the image to blob storage.
-- Update the partition layout on the board by running the following
+- [Prepare the Device](#prepare-the-device)
+  - [Install Prerequisites](#install-prerequisites)
+  - [Tag Your Device](#tag-your-device)
+  - [Prepare the Sample](#prepare-the-sample)
+  - [Build the Image](#build-the-image)
+  - [Flash the Image](#flash-the-image)
+- [Prepare the ADU Service](#prepare-the-adu-service)
+- [Deploy the Over the Air Update](#deploy-the-over-the-air-update)
+  - [Build the Update Image](#build-the-update-image)
+  - [Generate the ADU Update Manifest](#generate-the-adu-update-manifest)
+  - [Import the Update Manifest](#import-the-update-manifest)
+  - [Deploy Update](#deploy-update)
+
+**Note** that this sample does not utilize non-volatile storage of device credentials. That means that the update image contains the credentials themselves and each update may only apply to one device. For information on how to store credentials in non-volatile storage, please see the following link:
+
+- [ESP32 NVS API Documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/api-reference/storage/nvs_flash.html)
+- [ESP32 NVS Factory Flashing Tool](https://github.com/espressif/esp-idf/tree/master/tools/mass_mfg)
+
+## Prepare the Device
+
+### Install Prerequisites
+
+1. Git
+
+    Install `git` following the [official website](https://git-scm.com/).
+
+1. ESP-IDF
+
+    On Windows, install the ESPRESSIF ESP-IDF using this [download link](https://dl.espressif.com/dl/esp-idf/?idf=4.4). Please use ESP-IDF version 4.3 on Windows and version 4.4 on Linux.
+
+    For other Operating Systems or to update an existing installation, follow [Espressif official documentation](https://docs.espressif.com/projects/esp-idf/en/latest/esp32/get-started/index.html#get-started).
+
+1. [Powershell](https://docs.microsoft.com/shows/it-ops-talk/how-to-install-powershell-7)
```

Comment:
Would this be a better link here? The other one seems like it might disappear at some point
https://docs.microsoft.com/en-us/powershell/scripting/install/installing-powershell?view=powershell-7.2